dandi/validate.py

```diff
@@ -44,12 +90,101 @@ def validate_bids(
         )
+
     validation_result = validate_bids_(
         paths,
         schema_version=schema_version,
         report_path=report_path,
     )
-    return dict(validation_result)
+    our_validation_result = []
+    origin = ValidationOrigin(
+        name="bidsschematools",
+        version=bidsschematools.__version__,
+    )
+
+    for path in validation_result["path_tracking"]:
+        # Hard-coding exclusion here pending feature + release in:
+        # https://github.com/bids-standard/bids-specification/issues/1272
+        if path.endswith((".ERRORS", ".ERRORS.json")):
+            continue
+        dataset_path = _get_set_path(path, "dataset_description.json")
+        dandiset_path = _get_set_path(path, "dandiset.yaml")
+        our_validation_result.append(
+            ValidationResult(
+                origin=origin,
+                severity=Severity.ERROR,
+                id="BIDS.NON_BIDS_PATH_PLACEHOLDER",
+                scope=Scope.FILE,
+                path=Path(path),
+                message="File does not match any pattern known to BIDS.",
+                dataset_path=dataset_path,
+                dandiset_path=dandiset_path,
+            )
+        )
+
+    for pattern in validation_result["schema_tracking"]:
+        # Future proofing for standard-compliant name.
+        if pattern.get("mandatory") or pattern.get("required"):
+            # We don't have a path for this so we'll need some external logic to make sure
+            # that the dataset path is populated.
+            # dataset_path = _get_dataset_path(path, paths)
+            our_validation_result.append(
+                ValidationResult(
+                    origin=origin,
+                    severity=Severity.ERROR,
+                    id="BIDS.MANDATORY_FILE_MISSING_PLACEHOLDER",
+                    scope=Scope.DATASET,
+                    path_regex=pattern["regex"],
+                    message="BIDS-required file is not present.",
+                )
+            )
+    for meta in validation_result["match_listing"]:
+        file_path = meta.pop("path")
+        meta = {BIDS_TO_DANDI[k]: v for k, v in meta.items() if k in BIDS_TO_DANDI}
+        dataset_path = _get_set_path(file_path, "dataset_description.json")
+        dandiset_path = _get_set_path(file_path, "dandiset.yaml")
+        # Top level files do not have any other metadata other than path,
+        # which we pop and put in the object...
+        if not meta:
+            meta = {
+                "subject": None,
+            }
+        our_validation_result.append(
+            ValidationResult(
+                origin=origin,
+                id="BIDS.MATCH",
+                scope=Scope.FILE,
+                path=Path(file_path),
+                metadata=meta,
+                dataset_path=dataset_path,
+                dandiset_path=dandiset_path,
+            )
+        )
+
+    return our_validation_result
+
+
+def _get_set_path(in_path, marker):
```

Comment:
I had said before that we have already a function for such purposes and it should be used instead of reinventing the wheel!  Just use `find_parent_directory_containing` which is currently at https://github.com/dandi/dandi-cli/blob/HEAD/dandi/utils.py#L392

Reply:
290d1ed5df72f9de393f4c7390dee195a395a7fa